### PR TITLE
Sg/vertexid color

### PR DESCRIPTION
**Pull request by esmelusina**
---
### Purpose of this PR
Some minor feature work to expose vertex color as an interpolator target for Universal and HDRP. Vertex ID, Stereo Eye Index, and Instance ID are also exposed through nodes.

---
### Testing status
Tests added to ShaderGraph test for Color Interpolator and Vertex ID, this uses the Universal pipeline for testing. Instance ID was added to the Instancing tests in ShaderGraph-- these tests were also corrected to ensure instancing is active. Stereo Eye Index test was added to ShaderGraph Universal Stereo test (001). HDRP 8108 added to test vertex ID and color interpolator (which have unique code paths from URP) against a handwritten shader that does the equivalent.

Ref images and other yamato issues have (finally) been corrected. This PR is tested and ref images are updated.

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
I would like for HDRP team to add automated tests for this if possible, since it has unique code paths in HDRP material types. Other than that I think this is good. Instance ID doesn't always work correctly for me (on my computer it didn't work in play mode in my project, but it worked in the automated test.)

**Review by ValGrimm-U3D — APPROVED** [review]
Discussed two sentences to change with Esme, but didn't go deep into it as I don't have a lot of context. Don't see any major issues here. No need to hold this PR for the docs questionnaire, we'll do it after in https://unity3d.atlassian.net/browse/DOC-2690

**Comment by ValGrimm-U3D** [comment]
I'm not sure if your recent changes have involved doc but I can look at this next week.

**Review by LandonTownsendUnity — APPROVED** [review]
HDRP tests made it in so I think the PR is good now

**Review by pbbastian — CHANGES_REQUESTED** [review]
Looks good, only have one minor request

File: com.unity.shadergraph/Documentation~/Input-Nodes.md [hunk 1]
```diff
@@ -84,9 +90,9 @@
 |[**Scene Color**](Scene-Color-Node.md)|[**Scene Depth**](Scene-Depth-Node.md)|
 |![Image](images/SceneColorNodeThumb.png)|![Image](images/SceneDepthNodeThumb.png)|
 |Provides access to the current Camera's color buffer.|Provides access to the current Camera's depth buffer.|
-|[**Screen**](Screen-Node.md)||
-|![Image](images/ScreenNodeThumb.png)||
-|Provides access to parameters of the screen.||
+|[**Screen**](Screen-Node.md)|[**Eye Index**](Eye-Index-Node.md)|
+|![Image](images/ScreenNodeThumb.png)|![Image](images/EyeIndexNodeThumb.png)|
+|Provides access to parameters of the screen.|Provides access to the Eye Index during stereo rendering.|
```
Comment (ValGrimm-U3D) [inline com.unity.shadergraph/Documentation~/Input-Nodes.md:95]: "Provides access to screen parameters."

File: com.unity.shadergraph/Documentation~/Instance-ID-Node.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Instance ID Node
+
+## Description
+
+Provides access to the **Instance ID** assigned when rendering with GPU instancing.
```
Comment (ValGrimm-U3D) [inline com.unity.shadergraph/Documentation~/Instance-ID-Node.md:5]: "Provides access to the Instance ID that the GPU assigns when you render with GPU instancing."
(assigned is passive voice, so I am trying to fix it here)

**Review by ValGrimm-U3D — APPROVED** [review]
Suggested two minor docs changes @esmelusina . Accidentally started a review by clicking the wrong thing, so approving.

**Review by pbbastian — APPROVED** [review]

**Comment by esmelusina** [comment]
PR no longer necessary, color interpolator use-cases will be addressed by the Custom Interpolators feature here: https://github.com/Unity-Technologies/Graphics/pull/3281. The other utility nodes in this PR have already merged under a different PR.

**Pull request closed without merging**